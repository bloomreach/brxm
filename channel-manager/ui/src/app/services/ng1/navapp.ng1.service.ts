/*!
 * Copyright 2021 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { InjectionToken } from '@angular/core';
import { ClientError, NavLocation, ParentConfig, UserSettings } from '@bloomreach/navapp-communication';

export interface Ng1NavappService {
  subscribe: (api: string, callback: () => Promise<any>) => void;
  unsubscribe: (api: string, callback: () => Promise<any>) => void;
  connect: () => Promise<void>;
  getConfig(): Promise<ParentConfig>;
  getUserSettings(): Promise<UserSettings>;
  updateNavLocation(location: NavLocation): Promise<void>;
  navigate(location: NavLocation): Promise<void>;
  showMask(): Promise<void>;
  hideMask(): Promise<void>;
  showBusyIndicator(): Promise<void>;
  hideBusyIndicator(): Promise<void>;
  onUserActivity(): Promise<void>;
  onSessionExpired(): Promise<void>;
  onError(error: ClientError): Promise<void>;
}

export const NG1_NAVAPP_SERVICE = new InjectionToken<Ng1NavappService>('NG1_NAVAPP_SERVICE');
