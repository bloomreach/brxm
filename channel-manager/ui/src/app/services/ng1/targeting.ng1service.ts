/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { InjectionToken } from '@angular/core';

import { ExperimentGoal } from '../../experiments/models/experiment-goal.model';
import { ExperimentStatus } from '../../experiments/models/experiment-status.model';
import { Experiment } from '../../experiments/models/experiment.model';
import { TargetingApiResponse } from '../../models/targeting-api-response.model';
import { Variant, VariantCharacteristicData } from '../../variants/models/variant.model';

export interface Ng1TargetingService {
  getCharacteristics(): Promise<TargetingApiResponse<void>>;
  getVariants(containerItemId: string): Promise<TargetingApiResponse<Variant[]>>;
  addVariant(
    componentId: string,
    formData: any,
    personaId?: string,
    characteristics?: VariantCharacteristicData[],
  ): Promise<TargetingApiResponse<any>>;
  getGoals(): Promise<TargetingApiResponse<ExperimentGoal[]>>;
  getExperiment(componentId: string): Promise<TargetingApiResponse<Experiment>>;
  getExperimentStatus(experimentId: string): Promise<TargetingApiResponse<ExperimentStatus>>;
  saveExperiment(componentId: string, goalId: string, variantId: string): Promise<TargetingApiResponse<string>>;
  completeExperiment(componentId: string, keepOnlyVariantId: string): Promise<TargetingApiResponse<void>>;
}

export const NG1_TARGETING_SERVICE = new InjectionToken<Ng1TargetingService>('NG1_TARGETING_SERVICE');
