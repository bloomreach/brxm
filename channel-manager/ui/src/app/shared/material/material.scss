/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~@angular/material/theming';
@import '~@bloomreach/frontend-theme/palettes';
@import 'variables';

$custom-typography: mat-typography-config(
  $font-family: inspect($font-family)
);

@include mat-core($custom-typography);

$primary: mat-palette($blue);
$accent: mat-palette($orange);
$warn: mat-palette($red);
$theme: mat-light-theme($primary, $accent, $warn);

@include angular-material-theme($theme);

// default body text color
.mat-typography {
  color: map-get($text-color, "dark");
}
