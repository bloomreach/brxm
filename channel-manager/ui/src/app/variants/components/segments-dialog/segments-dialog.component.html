<!--
  Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
  -->

<div class="qa-segments-dialog mat-typography">
  <h3 mat-dialog-title>
    {{ 'DIALOG_SEGMENTS_TITLE' | translate }}
  </h3>

  <div mat-dialog-content>
    <mat-divider></mat-divider>
    <mat-list *ngIf="!personas || personas?.length === 0">
      <mat-list-item>
        {{ 'DIALOG_SEGMENTS_NO_SEGMENTS' | translate }}
      </mat-list-item>
    </mat-list>

    <mat-selection-list *ngIf="personas?.length > 0"
                        color="primary"
                        #personasList
                        [multiple]="false">
      <mat-list-option *ngFor="let persona of personas"
                       [value]="persona">
        {{ persona.name || persona.segmentName }}
      </mat-list-option>
    </mat-selection-list>

    <mat-divider></mat-divider>
  </div>

  <div mat-dialog-actions>
    <button mat-stroked-button
            type="button"
            [matDialogClose]>
      {{ 'CANCEL' | translate }}
    </button>
    <button mat-flat-button
            type="submit"
            color="primary"
            [disabled]="!hasSelectedPersona()"
            (click)="selectPersona()">
      {{ 'SELECT' | translate }}
    </button>
  </div>
</div>
