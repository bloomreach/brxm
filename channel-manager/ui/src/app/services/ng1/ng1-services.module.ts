/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { NgModule } from '@angular/core';

import { NG1_CHANNEL_SERVICE } from './channel.ng1service';
import { NG1_CONFIG_SERVICE } from './config.ng1.service';
import { NG1_CONTENT_SERVICE } from './content.ng1.service';
import { NG1_IFRAME_SERVICE } from './iframe.ng1service';
import { NG1_PAGE_STRUCTURE_SERVICE } from './page-structure.ng1.service';
import { NG1_PAGE_SERVICE } from './page.ng1.service';
import { NG1_PROJECT_SERVICE } from './project.ng1.service';
import { NG1_ROOT_SCOPE } from './root-scope.service';
import { NG1_WORKFLOW_SERVICE } from './workflow.ng1.service';

@NgModule({
  providers: [
    { provide: NG1_CHANNEL_SERVICE, useValue: window.angular.element(document.body).injector().get('ChannelService') },
    { provide: NG1_CONFIG_SERVICE, useValue: window.angular.element(document.body).injector().get('ConfigService') },
    { provide: NG1_CONTENT_SERVICE, useValue: window.angular.element(document.body).injector().get('ContentService') },
    { provide: NG1_IFRAME_SERVICE, useValue: window.angular.element(document.body).injector().get('HippoIframeService') },
    { provide: NG1_PAGE_STRUCTURE_SERVICE, useValue: window.angular.element(document.body).injector().get('PageStructureService') },
    { provide: NG1_PAGE_SERVICE, useValue: window.angular.element(document.body).injector().get('PageService') },
    { provide: NG1_PROJECT_SERVICE, useValue: window.angular.element(document.body).injector().get('ProjectService') },
    { provide: NG1_WORKFLOW_SERVICE, useValue: window.angular.element(document.body).injector().get('WorkflowService') },
    { provide: NG1_ROOT_SCOPE, useValue: window.angular.element(document.body).injector().get('$rootScope') },
  ],
})
export class Ng1ServicesModule {
}
