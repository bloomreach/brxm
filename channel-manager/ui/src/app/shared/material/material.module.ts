/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule, MatIconRegistry } from '@angular/material/icon';
import { MatListModule } from '@angular/material/list';
import { DomSanitizer } from '@angular/platform-browser';

import { registerSvgIcons } from './register-svg-icons';

@NgModule({
  exports: [
    MatButtonModule,
    MatListModule,
    MatIconModule,
  ],
})
export class MaterialModule {
  constructor(private readonly iconRegistry: MatIconRegistry, private readonly domSanitizer: DomSanitizer) {
    this.registerCustomIcons();
  }

  private registerCustomIcons(): void {
    registerSvgIcons(this.iconRegistry, this.domSanitizer);
  }
}
