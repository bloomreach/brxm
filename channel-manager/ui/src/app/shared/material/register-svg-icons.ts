/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// tslint:disable:match-default-export-name
import alertCircleOutline from '!!raw-loader!./icons/alert-circle-outline.svg';
import alertOutline from '!!raw-loader!./icons/alert-outline.svg';
import calendarClock from '!!raw-loader!./icons/calendar-clock.svg';
import chevronRight from '!!raw-loader!./icons/chevron-right.svg';
import commentCheckOutline from '!!raw-loader!./icons/comment-check-outline.svg';
import commentProcessingOutline from '!!raw-loader!./icons/comment-processing-outline.svg';
import commentRemoveOutline from '!!raw-loader!./icons/comment-remove-outline.svg';
import lockOutline from '!!raw-loader!./icons/lock-outline.svg';
import minusCircleOutline from '!!raw-loader!./icons/minus-circle-outline.svg';
import plus from '!!raw-loader!./icons/plus.svg';
import xpageIcon from '!!raw-loader!./icons/xpage.svg';
import { MatIconRegistry } from '@angular/material/icon';
import { DomSanitizer } from '@angular/platform-browser';

export function registerSvgIcons(iconRegistry: MatIconRegistry, donSanitizer: DomSanitizer): void {
  const registerIcon = (name: string, svg: string) => {
    iconRegistry.addSvgIconLiteral(name, donSanitizer.bypassSecurityTrustHtml(svg));
  };

  registerIcon('xpage', xpageIcon);
  registerIcon('minus-circle-outline', minusCircleOutline);
  registerIcon('alert-outline', alertOutline);
  registerIcon('comment-processing-outline', commentProcessingOutline);
  registerIcon('comment-remove-outline', commentRemoveOutline);
  registerIcon('calendar-clock', calendarClock);
  registerIcon('alert-circle-outline', alertCircleOutline);
  registerIcon('lock-outline', lockOutline);
  registerIcon('comment-check-outline', commentCheckOutline);
  registerIcon('plus', plus);
  registerIcon('chevron-right', chevronRight);
}
