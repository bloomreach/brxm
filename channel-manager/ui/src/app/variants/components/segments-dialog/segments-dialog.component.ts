/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Component, Inject, OnInit, ViewChild } from '@angular/core';
import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
import { MatSelectionList } from '@angular/material/list';

import { Ng1TargetingService, NG1_TARGETING_SERVICE } from '../../../services/ng1/targeting.ng1service';
import { Persona } from '../../models/persona.model';

@Component({
  selector: 'em-segments-dialog',
  templateUrl: './segments-dialog.component.html',
  styleUrls: ['./segments-dialog.component.scss'],
})
export class SegmentsDialogComponent implements OnInit {
  personas?: Persona[];

  @ViewChild('personasList')
  selectionList?: MatSelectionList;

  constructor(
      @Inject(NG1_TARGETING_SERVICE) private readonly targetingService: Ng1TargetingService,
      @Inject(MAT_DIALOG_DATA) public data: any,
      private readonly dialogRef: MatDialogRef<SegmentsDialogComponent>,
  ) { }

  async ngOnInit(): Promise<void> {
    const response = await this.targetingService.getPersonas();
    this.personas = response.data.items
      .filter(persona => !this.data.selectedPersonaIds.includes(persona.id));
  }

  selectPersona(): void {
    const selectedPersona = this.selectionList?.selectedOptions.selected[0].value;
    this.dialogRef.close(selectedPersona);
  }

  hasSelectedPersona(): boolean {
    return this.selectionList?.selectedOptions.hasValue() === true;
  }
}
