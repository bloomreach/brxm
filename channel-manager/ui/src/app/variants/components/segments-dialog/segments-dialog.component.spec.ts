/*
 * Copyright 2020-2023 Bloomreach
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatDialogModule, MatDialogRef } from '@angular/material/dialog';
import { MatListModule } from '@angular/material/list';
import { TranslateModule } from '@ngx-translate/core';

import { TargetingApiResponse } from '../../../models/targeting-api-response.model';
import { Ng1TargetingService, NG1_TARGETING_SERVICE } from '../../../services/ng1/targeting.ng1.service';
import { Persona } from '../../models/persona.model';

import { SegmentsDialogComponent } from './segments-dialog.component';

describe('SegmentsDialogComponent', () => {
  let component: SegmentsDialogComponent;
  let fixture: ComponentFixture<SegmentsDialogComponent>;

  let targetingService: Ng1TargetingService;

  const personas: Partial<Persona>[] = [
    { segmentName: 'Walmart' },
    { segmentName: 'Cosco' },
    { segmentName: 'Vons' },
    { segmentName: '7eleven' },
  ];

  const mockApiResponse: TargetingApiResponse<{ items: Persona[] }> = {
    success: true,
    message: 'Some message',
    errorCode: null,
    reloadRequired: false,
    data: {
      items: personas as Persona[],
    },
  };

  beforeEach(() => {
    const targetingServiceMock = {
      getPersonas: jest.fn(() => Promise.resolve(mockApiResponse)),
    };

    const matDialogRefMock = {};

    TestBed.configureTestingModule({
      imports: [
        MatDialogModule,
        MatListModule,
        TranslateModule.forRoot(),
      ],
      declarations: [SegmentsDialogComponent],
      providers: [
        { provide: NG1_TARGETING_SERVICE, useValue: targetingServiceMock },
        { provide: MatDialogRef, useValue: matDialogRefMock },
      ],
    });

    targetingService = TestBed.inject(NG1_TARGETING_SERVICE);
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(SegmentsDialogComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  describe('segments list', () => {
    it('should render personas list alphabetically', async () => {
      await component.ngOnInit();
      fixture.detectChanges();

      const segmentListElement = fixture.nativeElement.querySelector('.qa-segments');
      expect(segmentListElement).toMatchSnapshot();
    });
  });
});
