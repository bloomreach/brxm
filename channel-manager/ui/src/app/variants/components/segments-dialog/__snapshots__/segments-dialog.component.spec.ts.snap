// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentsDialogComponent segments list should render personas list alphabetically 1`] = `
<mat-selection-list
  aria-disabled="false"
  aria-multiselectable="false"
  class="qa-segments mat-selection-list mat-list-base"
  color="primary"
  role="listbox"
  tabindex="0"
>
  
  <mat-list-option
    aria-disabled="false"
    aria-selected="false"
    class="mat-list-item mat-list-option mat-focus-indicator mat-primary"
    role="option"
    tabindex="-1"
  >
    <div
      class="mat-list-item-content mat-list-item-content-reverse"
    >
      <div
        class="mat-list-item-ripple mat-ripple"
        mat-ripple=""
      />
      
      <div
        class="mat-list-text"
      >
         7eleven 
      </div>
    </div>
  </mat-list-option>
  <mat-list-option
    aria-disabled="false"
    aria-selected="false"
    class="mat-list-item mat-list-option mat-focus-indicator mat-primary"
    role="option"
    tabindex="-1"
  >
    <div
      class="mat-list-item-content mat-list-item-content-reverse"
    >
      <div
        class="mat-list-item-ripple mat-ripple"
        mat-ripple=""
      />
      
      <div
        class="mat-list-text"
      >
         Cosco 
      </div>
    </div>
  </mat-list-option>
  <mat-list-option
    aria-disabled="false"
    aria-selected="false"
    class="mat-list-item mat-list-option mat-focus-indicator mat-primary"
    role="option"
    tabindex="-1"
  >
    <div
      class="mat-list-item-content mat-list-item-content-reverse"
    >
      <div
        class="mat-list-item-ripple mat-ripple"
        mat-ripple=""
      />
      
      <div
        class="mat-list-text"
      >
         Vons 
      </div>
    </div>
  </mat-list-option>
  <mat-list-option
    aria-disabled="false"
    aria-selected="false"
    class="mat-list-item mat-list-option mat-focus-indicator mat-primary"
    role="option"
    tabindex="-1"
  >
    <div
      class="mat-list-item-content mat-list-item-content-reverse"
    >
      <div
        class="mat-list-item-ripple mat-ripple"
        mat-ripple=""
      />
      
      <div
        class="mat-list-text"
      >
         Walmart 
      </div>
    </div>
  </mat-list-option>
</mat-selection-list>
`;
